test/test_dblib.ml

```diff
@@ -28,8 +28,9 @@ let with_conn (user, password, server, database) f =
   let conn = Dblib.connect ~user ~password server in
   Dblib.use conn database;
   try
-    f conn;
-    Dblib.close conn
+    let res = f conn in
+    Dblib.close conn;
+    res
```

Comment:
Do not see the point of this modification.

Reply:
Previously `with_conn` always returned `unit`, but in the new `test_concurrency` function it needs to return `(unit, exn) Result.t`